msg/message.go

```diff
@@ -32,24 +33,12 @@ const (
 	// NotImplemented occurs when a message or request is received whos response
 	// requires functionality that does not yet exist.
 	NotImplemented = 501
-	// SubnetFull occurs when a stream is opened with a peer whose Subnet is
-	// already full.
-	SubnetFull = 503
 )
 
-// Initializes all the types we need to encode.
-func init() {
-	gob.Register(&Request{})
-	gob.Register(&Response{})
-	gob.Register(&Push{})
-	gob.Register(&blockchain.Transaction{})
-	gob.Register(&blockchain.Block{})
-}
-
 // ProtocolError is an error that occured during a request.
 type ProtocolError struct {
-	Code    ErrorCode
-	Message string
+	Code    ErrorCode `json:"Code"`
```

Comment:
If the json struct tag is the same as the struct field name, then it can be omitted. I'd say either remove the struct tags or make them `camelCase` like typical JSON